### PR TITLE
seekerArray argument, test, and vignette updates

### PR DESCRIPTION
**Pull request by JSchoenbachler**
- `seekerArray` now accepts either a named list (how it previously worked) or individual arguments.
- Vignettes, included scripts, documentation, and tests are updated to reflect this.
- `getProbeGeneMapping` test now checks the version of the mapping package being used in the test. If it doesn't match the version used to generate the test, we skip the test.
- Additionally, if the mapping package version is newer than the version used to generate the snapshot, we provide a warning to prompt the user to regenerate the test.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# seeker 1.0.3
+* Make `getProbeGeneMapping` test check for version number of mapping package.
```
Comment (jakejh) [inline NEWS.md]: Use consistent (past) tense and add `()` for functions.

File: R/array.R [hunk 2]
```diff
@@ -93,8 +88,15 @@ checkSeekerArrayArgs = function(params, parentDir) {
 #' @seealso [seeker()]
 #'
 #' @export
-seekerArray = function(params, parentDir = '.') {
-  r = checkSeekerArrayArgs(params, parentDir)
+seekerArray = function(study, geneIdType, platform = NULL, params = NULL, parentDir = '.') {
```
Comment (jakejh) [inline R/array.R]: Let's just ditch the `params` argument. I think that's simpler.

File: R/array.R [hunk 3]
```diff
@@ -93,8 +88,15 @@ checkSeekerArrayArgs = function(params, parentDir) {
...
+  if (!is.null(params)) {
+    study = params$study
+    geneIdType = params$geneIdType
+    platform = params$platform
+  } else {
+    params = list(study = study, geneIdType = geneIdType, platform = platform)
```
Comment (jakejh) [inline R/array.R]: Irrelevant if we remove `params`, but I don't see how this code has any effect.

File: inst/extdata/run_seeker_array.sh [hunk 4]
```diff
@@ -5,5 +5,5 @@ docker run \
   -w /home/rstudio/projects \
   --rm \
   ghcr.io/hugheylab/socker \
-  bash -c "Rscript run_seeker_array.R GSE25585.yml ." \
+  bash -c "Rscript -e "'"'"seeker::seekerArray('GSE25585', 'ensembl', parentDir = '.')"'"' \
```
Comment (jakejh) [inline inst/extdata/run_seeker_array.sh]: Nope, too opaque. I still want the shell script to run an R script that loads a yaml file.

File: tests/testthat/test-array-utils.R [hunk 5]
```diff
@@ -162,6 +162,13 @@ test_that('getEntrezEnsemblMapping', {
 
 
 test_that('getProbeGeneMapping', {
+  mapPkgVerObs = packageVersion('org.Mm.eg.db')
+  mapPkgVerExp = snapshot(mapPkgVerObs, file.path(dataDir, 'map_pkg_ver.qs'))
+  if (mapPkgVerObs != mapPkgVerExp) {
+    if (mapPkgVerObs < mapPkgVerExp) warning('New version of map package available, test snapshots should be regenerated.')
```
Comment (jakejh) [inline tests/testthat/test-array-utils.R]: fix line length

File: tests/testthat/test-array.R [hunk 6]
```diff
@@ -6,7 +6,7 @@ withr::local_file(parentDirArr)
 test_that('checkSeekerArrayArgs', {
   skip_on_os('windows', arch = NULL)
 
-  resultObs = checkSeekerArrayArgs(paramsArray, parentDirArr)
+  resultObs = checkSeekerArrayArgs(paramsArray$study, paramsArray$geneIdType, paramsArray$platform, parentDirArr)
```
Comment (jakejh) [inline tests/testthat/test-array.R]: line length

**Review by jakejh — CHANGES_REQUESTED** [review]
Make sure to revise vignettes according to my requested changes.

**Comment by JSchoenbachler** [comment]
Resolved most items with https://github.com/hugheylab/seeker/pull/34/commits/e5b997d7aadba8bdb9a63caeea3fec30787f23f8

File: inst/extdata/run_seeker_array.sh (hunk 4 above)
Comment (JSchoenbachler) [reply inst/extdata/run_seeker_array.sh]: Look at the current version. Would you rather it load from yaml or use the way I have specifying each argument?

Comment (jakejh) [reply inst/extdata/run_seeker_array.sh]: I think we can show both.

File: DESCRIPTION [hunk 7]
```diff
@@ -1,10 +1,11 @@
 Package: seeker
 Type: Package
 Title: Process Genomic Data
-Version: 1.0.2
+Version: 1.0.3
 Authors@R: c(
-  person('Jake', 'Hughey', , 'jakejhughey@gmail.com', c('aut', 'cre')),
-  person('Josh', 'Schoenbachler', , 'josh.schoenbachler@gmail.com', 'aut'))
+    person("Jake", "Hughey", , "josh.schoenbachler@gmail.com", role = c("aut", "cre")),
```
Comment (jakejh) [inline DESCRIPTION]: Fix email address. Was it necessary to change indentation and quotation marks?

File: tests/testthat/test-array.R [hunk 8]
```diff
@@ -6,7 +6,8 @@ withr::local_file(parentDirArr)
 test_that('checkSeekerArrayArgs', {
   skip_on_os('windows', arch = NULL)
 
-  resultObs = checkSeekerArrayArgs(paramsArray, parentDirArr)
+  resultObs = checkSeekerArrayArgs(paramsArray$study, paramsArray$geneIdType,
```
Comment (jakejh) [inline tests/testthat/test-array.R]: There's a better fix for line length.

File: tests/testthat/test-array.R [hunk 9]
```diff
@@ -16,26 +17,24 @@ test_that('checkSeekerArrayArgs', {
 test_that('checkSeekerArrayArgs errors', {
   skip_on_os('windows', arch = NULL)
 
-  paramsArrayErr = paramsArray
-
   # GSE platform not null or GPL
-  paramsArrayErr$platform = 'abcd'
-  expect_error(checkSeekerArrayArgs(paramsArrayErr, parentDirArr))
+  expect_error(checkSeekerArrayArgs(paramsArray$study, paramsArray$geneIdType,
```
Comment (jakejh) [inline tests/testthat/test-array.R]: Better, simpler fix for line length that we've used before.

File: tests/testthat/test-array-utils.R [hunk 10]
```diff
@@ -162,6 +162,14 @@ test_that('getEntrezEnsemblMapping', {
 
 
 test_that('getProbeGeneMapping', {
+  mapPkgVerObs = packageVersion('org.Mm.eg.db')
+  mapPkgVerExp = snapshot(mapPkgVerObs, file.path(dataDir, 'map_pkg_ver.qs'))
+  if (mapPkgVerObs != mapPkgVerExp) {
+    if (mapPkgVerObs < mapPkgVerExp) warning('New version of map package available,
```
Comment (jakejh) [inline tests/testthat/test-array-utils.R]: I'm going to keep harping on line length.

File: inst/extdata/run_seeker_array.R [hunk 11]
```diff
@@ -1,6 +1,20 @@
 cArgs = commandArgs(TRUE)
-yamlPath = cArgs[1L]
-parentDir = cArgs[2L]
+if (length(cArgs) == 2) {
```
Comment (jakejh) [inline inst/extdata/run_seeker_array.R]: Too complicated. I still think the R script should just load a yaml file and the shell script should not change. I don't think anything in the reproducibility vignette has to change. Only one line in run_seeker_array.R needs to change.

**Review by jakejh — CHANGES_REQUESTED** [review]
Not quite there yet.

File: DESCRIPTION (hunk 7 above)
Comment (JSchoenbachler) [reply DESCRIPTION]: I don't remember changing this, nor do I think I would have had any reason to. Is it possible that documenting the package or another step could have changed/updated it?

**Comment by JSchoenbachler** [comment]
@jakejh working on your other items, but did you see the test failures with the "Ae" tests? See the GH Actions run here: https://github.com/hugheylab/seeker/runs/7007574149?check_suite_focus=true

**Comment by JSchoenbachler** [comment]
Resolved your issues with https://github.com/hugheylab/seeker/pull/34/commits/9a3a30a9aec41e695950b20a3e0e49eafa8bbd83

**Comment by JSchoenbachler** [comment]
Looks like the test issue with "AE" studies was a one-off.

File: DESCRIPTION (hunk 7 above)
Comment (jakejh) [reply DESCRIPTION]: Email address still not fixed. Please be careful and do not rush.

File: inst/extdata/run_seeker_array.R (hunk 11 above)
Comment (jakejh) [reply inst/extdata/run_seeker_array.R]: As requested, please restore the reproducibility vignette to the version on master.

Comment (JSchoenbachler) [reply inst/extdata/run_seeker_array.R]: Done in https://github.com/hugheylab/seeker/pull/34/commits/1774d453dc5f18373968ee483886a7bfe56b7f10

File: DESCRIPTION (hunk 7 above)
Comment (JSchoenbachler) [reply DESCRIPTION]: Resolved with https://github.com/hugheylab/seeker/pull/34/commits/1774d453dc5f18373968ee483886a7bfe56b7f10

**Comment by JSchoenbachler** [comment]
Figured out what happened with the email.

When running a `devtools::check_win_*()` function and specifying the `email` argument, the function will modify the `DESCRIPTION` file and set the maintainer's email to the email you specify (since I guess win-builder just emails the maintainer the results). Normally, it will overwrite the file back to it's original state once the function finishes, but if you have the `DESCRIPTION` file open, it wont revert back unless you close and reopen the file. I guess I left it open, changed the version number, and that kept the change.

**Comment by JSchoenbachler** [comment]
@jakejh I think all changes have been made. The only thing remaining is the inclusion of examples. Do you want to have those be a part of this pull request? Or just finish up this pull request, resubmit to cran, and see what happens?

**Review by jakejh — APPROVED** [review]

**Pull request merged**